consensus/src/tests/dag_tests.rs

```diff
@@ -56,8 +56,89 @@ async fn test_dag_new_has_genesis() {
     let (_tx_cert, rx_cert) = channel(1);
     let (_, dag) = Dag::new(&committee, rx_cert);
 
+    for certificate in genesis.clone() {
+        assert!(dag.contains(certificate).await);
+    }
+
+    // But the genesis does not come out in read_causal, as is is compressed the moment we add more nodes
+    let (certificates, _next_parents) = make_optimal_certificates(1..=1, &genesis, &keys);
+    let mut certs_to_insert = certificates.clone();
+
+    // Feed the additional certificates to the Dag
+    while let Some(certificate) = certs_to_insert.pop_front() {
+        dag.insert(certificate).await.unwrap();
+    }
+
+    // genesis is still here
+    for certificate in genesis.clone() {
+        assert!(dag.contains(certificate).await);
+    }
+
+    // we trigger read_causal on the newly inserted cert
+    for cert in certificates.clone() {
+        let res = dag.read_causal(cert.digest()).await.unwrap();
+        // the read_causals do not report genesis: we only walk one node, the start of the walk
+        assert_eq!(res, vec![cert.digest()]);
+    }
+
+    // genesis is no longer here
     for certificate in genesis {
-        assert!(dag.contains(certificate).await)
+        assert!(!dag.contains(certificate).await);
+    }
+}
+
+// The above test relies on the fact that genesis produces empty blocks: we re-run it with non-genesis empty blocks to
```

Comment:
Nit: `The above test...`  ...I would suggest to refer to the name directly `test_dag_new_has_genesis_and_its_not_live` , it's always possible someone inserting a new test between those two and then introduce some confusion